src/azure-cli/azure/cli/command_modules/botservice/custom.py

```diff
@@ -74,15 +75,27 @@ def __prepare_configuration_file(cmd, resource_group_name, kudu_client, folder_p
             f.write(json.dumps(existing))
 
 
-def create(cmd, client, resource_group_name, resource_name, kind, msa_app_id, password, language=None,  # pylint: disable=too-many-locals, too-many-statements
+def create(cmd, client, resource_group_name, resource_name, kind, msa_app_id, password=None, language=None,  # pylint: disable=too-many-locals, too-many-statements, inconsistent-return-statements
            description=None, display_name=None, endpoint=None, tags=None, location='Central US',
-           sku_name='F0', version='v4', deploy_echo=None):
+           sku_name='F0', deploy_echo=None):
     # Kind parameter validation
     kind = kind.lower()
     registration_kind = 'registration'
     bot_kind = 'bot'
     webapp_kind = 'webapp'
 
+    # Mapping: registration is deprecated, we now use 'bot' kind for registration bots
+    if kind == registration_kind:
+        kind = bot_kind
+
+    # Check the resource name availability for the bot.
+    name_response = NameAvailability.check_name_availability(client, resource_name, kind)
+    if not name_response.valid:
+        # If the name is unavailable, gracefully exit and log the reason for the user.
```

Comment:
Should we throw exception here?

Reply:
That makes much more sense, I'll change the code to do that instead.